lib/web_ui/lib/src/engine/text_editing.dart

```diff
@@ -305,6 +308,18 @@ class TextEditingElement {
     _subscriptions
       ..add(html.document.onSelectionChange.listen(_handleChange))
       ..add(domElement.onInput.listen(_handleChange));
+
+    // In Firefox, when cursor moves, nor selectionChange neither onInput
+    // events are triggered. We are listening to keyup event to decide
+    // if the user shifted the cursor.
+    // See [_SelectionChangeDetection].
+    if (browserEngine == BrowserEngine.firefox) {
+      _subscriptions.add(domElement.onKeyUp.listen((event) {
+        if (_selectionDetection.hasChangeDetected(domElement)) {
```

Comment:
To make sure that we always use the same `domElement`, let's make it a constructor argument.
```dart
final SelectionChangeDetection _selectionDetection = SelectionChangeDetection(domElement);
...
if (_selectionDetection.hasChangeDetected()) { ... }
```

Reply:
Done.